numpy/linalg/linalg.py

```diff
@@ -148,8 +152,9 @@ def _commonType(*arrays):
                         (a.dtype.name,))
         else:
             rt = double
-        if rt is double:
-            result_type = double
+        if dtype_priority[rt] > dtype_priority[result_type]:
```

Comment:
I think you could just compare `rt.itemsize > result_type.itemsize`, and lose the dict

Reply:
@eric-wieser I have made the changes that you suggested by using itemsize and dropping the dict. Could you please guide me as to where I need to write the appropriate tests and release notes.

Reply:
Tests go in `linalg/tests/test_linalg.py`. I'd like to see at least a test that the upcasting to float64, computing the result, and then downcasting back to float16 works as expected (so check output type and result accuracy).